homeassistant/components/bosch_shc/entity.py

```diff
@@ -0,0 +1,112 @@
+"""Bosch Smart Home Controller base entity."""
+from boschshcpy.device import SHCDevice
+
+from homeassistant.helpers.device_registry import async_get_registry as get_dev_reg
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_registry import async_get_registry as get_ent_reg
+
+from .const import DOMAIN
+
+
+async def get_device_id(hass, device_id):
+    """Get device id from device registry."""
+    dev_registry = await get_dev_reg(hass)
+    device = dev_registry.async_get_device(
+        identifiers={(DOMAIN, device_id)}, connections=set()
+    )
+    if device is None:
+        return None
+    return device.id
+
+
+async def remove_devices(hass, entity, entry_id):
+    """Get item that is removed from session."""
+    await entity.async_remove()
+    ent_registry = await get_ent_reg(hass)
+    if entity.entity_id in ent_registry.entities:
+        ent_registry.async_remove(entity.entity_id)
```

Comment:
If we remove the entity entry from the entity registry the `Entity` instance will be removed automatically.

Reply:
Ok, this means I can remove the `entity.async_remove()` call?

Reply:
I think so, yes.

Reply:
I removed the `entity.async_remove()` call.